File: materials/sort/insertionSort.java
public static void insertionSort(int[] a)
	{
		for(int i = 1; i < a.length; ++i)
		{
			// invariant: x[0...i-1] is sorted
			// goal: shift x[i] down to its proper place: x[0...i]
			
			int j = i;
			int t = a[j];

			while(j > 0 && a[j-1] > t)
			{
				a[j] = a[j-1];
				--j;
			}
			
			a[j] = t;
		}
	}

File: materials/sort/insertionSort.js
function insertionSort(a) {
  for (let i = 1; i < a.length; i++) {

    let j = i;
    let t = a[j];

    while (j > 0 && a[j - 1] > t) {
      a[j] = a[j - 1];
      j--;
    }
    a[j] = t;
  }
}

// test
// let arr = [1,4,24,3,2,1,6,7,164,4,0,0,0,13];
// insertionSort(arr);
// console.log(arr);